snake-2/snake.js

```diff
@@ -0,0 +1,307 @@
+
+window.onload = () => createSnakeEngine(
+    "snakeCanvas", 
+    "foodCounter",
+    20,
+    200,
+    30,
+    getSnakeCoordinates);
+
+
+var snake = {
+    x: 0,
+    y: 0,
+    direction: "down"
+}
+
+
+function getSnakeCoordinates(mapSize, foodCoordinates) {
+   
+    let closestFoodIndices = findClosestFood(snake, foodCoordinates);
+    newHeading(snake, closestFoodIndices);
+
+    if(snake.direction == "down") {
+        var newY = snake.y + 1;
+
+        if(newY >= mapSize) {
+            snake.direction = "right";
+        } else {
+            snake.y = newY;
+        }
+    }
+
+    if(snake.direction == "right") {
+        var newX = snake.x + 1;
+
+        if(newX >= mapSize) {
+            snake.direction = "up";
+        } else {
+            snake.x = newX;
+        }
+    }
+
+    if(snake.direction == "up") {
+        var newY = snake.y - 1;
+
+        if(newY < 0) {
+            snake.direction = "left";
+        } else {
+            snake.y = newY;
+        }
+    }
+
+    if(snake.direction == "left") {
+        var newX = snake.x - 1;
+
+        if(newX < 0) {
+            snake.direction = "down";
+        } else {
+            snake.x = newX;
+        }
+    }
+
+    return [{
+        x: snake.x,
+        y: snake.y
+    }];
+
+}
+
+function findClosestFood(snake, foodCoordinates) {
```

Comment:
pytanko bonusowe - czy musimy zawsze analizować wszystkie współrzędne jedzenia? Czy istnieje jakaś sytuacja w której będziemy wiedzieć na pewno, że niczego bliżej już nie znajdziemy?

Reply:
Wydaje mi się, że żeby być pewnym, że niczego bliżej nie znajdziemy, trzeba by sprawdzać, czy nie ma jedzenie w pobliżu snake'a (zaczynając od pól o współrzędnych o krok od niego, potem o dwa, trzy itp...) – próbuję teraz coś takiego zrobić, jak się uda, to wyślę jako alternatywne rozwiązanie.

Dzięki za sprawdzenie! :)